libs/community/langchain_community/document_loaders/dedoc.py

```diff
@@ -0,0 +1,416 @@
+from abc import ABC, abstractmethod
+from typing import (
+    Dict,
+    Iterator,
+    Optional,
+    Tuple,
+    Union,
+)
+
+from langchain_core.documents import Document
+
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class DedocBaseLoader(BaseLoader, ABC):
```

Comment:
I'd recommend implementing use a BlobParser which will have access to mimetype information and it'll plug in nicely into the existing blob generators.

See here: 
https://python.langchain.com/v0.1/docs/modules/data_connection/document_loaders/custom/#baseblobparser

Reply:
We were thinking about implementing a `BlobParser`, but, in this case, it won't be always applicable.
For example:
```python
blob = Blob(data=b"some text")
```
Here `blob.path = None` and `blob.mimetype = None` and `dedoc` can't parse the content.

The usage of `dedoc` is similar to [unstructured](https://github.com/Unstructured-IO/unstructured) library, it is also used in `langchain` in document loaders only (e.g. [UnstructuredFileLoader](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/document_loaders/unstructured.py)).

Would it be appropriate if we don't implement a `BlobParser`?

Reply:
You're free to keep this as is, though I'd recommend using a blob parser to avoid repeating design mistakes in existing loaders. 

----

Many of the document loaders make an unnecessary assumption about the files having to existing on the local file system. This makes them more difficult to use in a production setting as content can't just be loaded from s3 / google storage into memory, but has to be duplicated to the local file system first.

If you create a blob parser, it allows for the content to be loaded from places other than the file system -- you're still free to wrap it in a document loader if you'd like.

As a blob parser users will be able to also use it with existing blob loaders that allow loading files from directories on the local file system or from cloud storage

https://python.langchain.com/v0.2/docs/how_to/document_loader_custom/#generic-loader

https://api.python.langchain.com/en/latest/document_loaders/langchain_community.document_loaders.blob_loaders.cloud_blob_loader.CloudBlobLoader.html#langchain_community.document_loaders.blob_loaders.cloud_blob_loader.CloudBlobLoader

Reply:
Let me know what you decide

Reply:
We consulted and decided to keep this as is